osquery/tables/system/linux/system_info.cpp

```diff
@@ -40,15 +41,12 @@ QueryData genSystemInfo(QueryContext& context) {
     }
   }
 
-  // Can parse /proc/cpuinfo or /proc/meminfo for this data.
-  static long cores = sysconf(_SC_NPROCESSORS_CONF);
-  if (cores > 0) {
-    r["cpu_logical_cores"] = INTEGER(cores);
-    r["cpu_physical_cores"] = INTEGER(cores);
-  } else {
-    r["cpu_logical_cores"] = "-1";
-    r["cpu_physical_cores"] = "-1";
-  }
+  auto logical_cores = boost::thread::hardware_concurrency();
```

Comment:
**static** 
It will protect us from doing syscall many times.

Reply:
So it looks like the boost API dynamically picks changes to the available logical cores.  So if we were to say disable hyperthreading within the life of an osquery process, a new query wouldn't be able to pick up the change if it were `static` if there was a query made before the change.  Is this what we would want for this particular column?  The column description in the table spec is looks like this ```Column("cpu_logical_cores", INTEGER, "Max number of CPU logical cores"),```.  Should we this column be actually just the the max possible logical cores based the hardware and there should be another column called `cpu_active_logical_cores`?  Or should `cpu_logical_cores`  be the current active logical cores of the system and we should augment the the table with an additional column `cpu_max_logical_cores`?   I ask because we will be needing both data points at some point :)

Reply:
Good point. The first thing we should we aware of is the backward compatibility. People might be using things in a way we are not aware of. 

I would vote for, reseting change in "cpu_logical_cores" as it was correct and only fixing physical_cores in this PR. We can discuss the future of the cpu_logical_cores in other issue or the PR.

Reply:
The documentation for these two columns is just wrong. We can either fix the implementation or the documentation. My feeling is that people expect this to be the number of cores in the system regardless of what the documentation says, so I would be for fixing the documentation here and keep your current behaviour. Later we can discuss introducing new columns.

Reply:
@allanliu 
Looks like STL also has this functionality. https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency
We should be good to merge after this change. 👍  Thanks